src/main/scala/rocket/ICache.scala

```diff
@@ -382,28 +731,44 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
               && s2_valid && s2_data_decoded.error && !s2_tag_disparity) {
           // handle correctable errors on CPU accesses to the scratchpad.
           // if there is an in-flight slave-port access to the scratchpad,
-          // report the a miss but don't correct the error (as there is
+          // report the miss but don't correct the error (as there is
           // a structural hazard on s1s3_slaveData/s1s3_slaveAddress).
           s3_slaveValid := true
           s1s3_slaveData := s2_data_decoded.corrected
           s1s3_slaveAddr := s2_scratchpad_word_addr | s1s3_slaveAddr(log2Ceil(wordBits/8)-1, 0)
         }
 
+        // back pressure is allowed on the [[tl]]
+        // pull up [[respValid]] when [[s2_slaveValid]] until [[tl.d.fire()]]
         respValid := s2_slaveValid || (respValid && !tl.d.ready)
+        // if [[s2_full_word_write]] will overwrite data, and [[s2_data_decoded.uncorrectable]] can be ignored.
         val respError = RegEnable(s2_scratchpad_hit && s2_data_decoded.uncorrectable && !s1s2_full_word_write, s2_slaveValid)
         when (s2_slaveValid) {
+          // need stage 3 if Put or correct decoding.
+          // @todo if uncorrectable [[s2_data_decoded]]?
           when (edge_in.get.hasData(s1_a) || s2_data_decoded.error) { s3_slaveValid := true }
+          /** data not masked by the TileLink PutData/PutPartitalData.
+            * means data is stored at [[s1s3_slaveData]] which was read at stage 1.
+            */
           def byteEn(i: Int) = !(edge_in.get.hasData(s1_a) && s1_a.mask(i))
+          // write [[s1s3_slaveData]] based on index of wordBits.
+          // @todo seems a problem here?
+          //       granularity of CPU fetch is `wordBits/8`,
+          //       granularity of TileLink access is `TLBundleParameters.dataBits/8`
+          //       these two granularity can be different.
+          // store data read from RAM
           s1s3_slaveData := (0 until wordBits/8).map(i => Mux(byteEn(i), s2_data_decoded.corrected, s1s3_slaveData)(8*(i+1)-1, 8*i)).asUInt
         }
 
         tl.d.valid := respValid
         tl.d.bits := Mux(edge_in.get.hasData(s1_a),
+          // PutData/PutPartialData -> AccessAck
           edge_in.get.AccessAck(s1_a),
+          // Get -> AccessAckData
           edge_in.get.AccessAck(s1_a, UInt(0), denied = Bool(false), corrupt = respError))
+        // @todo why directly `edge_in.get.AccessAck(s1_a, s1s3_slaveData, denied = Bool(false), corrupt = respError)`
         tl.d.bits.data := s1s3_slaveData
 
-        // Tie off unused channels
         tl.b.valid := false
```

Comment:
Why remove this comment?

Reply:
I have removed this line for conciseness. 
But I guess there is no bad to keep it? I will have it restored.